javalib/src/main/scala/java/util/HashSet.scala

```diff
@@ -0,0 +1,89 @@
+package java.util
+
+import scala.collection.mutable
+import scala.collection.JavaConversions._
+
+class HashSet[E](initialCapacity: Int, loadFactor: Float) 
+    extends java.util.AbstractSet[E] with java.util.Set[E] with Cloneable with Serializable {
+
+  def this() =
+  	this(16, 0.75F)
+
+  def this(initialCapacity: Int) =
+  	this(initialCapacity, 0.75F)
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  protected val inner: mutable.Map[Int, E] = new scala.collection.mutable.HashMap[Int, E]()
+
+  override def contains(o: Any): Boolean =
+    inner.contains(o.hashCode)
+
+  override def remove(o: Any): Boolean =
+    inner.remove(o.hashCode).isDefined
+
+  override def containsAll(c: Collection[_]): Boolean =
+    asScalaIterator(c.iterator).forall(e => inner.contains(e.hashCode))
+
+  override def removeAll(c: Collection[_]): Boolean =
+    removeWhere(c)(o => inner.contains(o.hashCode))
+
+  override def retainAll(c: Collection[_]): Boolean =
+    removeWhere(c)(o => !inner.contains(o.hashCode))
+
+  override def add(e: E): Boolean = {
+    val hc = e.hashCode
+    if (!inner.contains(hc)) {
+      inner += (e.hashCode -> e)
+      true
+    } else false
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+  	val iter = c.iterator()
+    var changed = false
+    while (iter.hasNext)
+      changed = add(iter.next()) || changed
+    changed
+  }  
+
+  override def size(): Int = inner.size
+
+  def iterator(): Iterator[E] =
+    new Iterator[E] {
+
+      private val keysCopy = inner.keysIterator
+      
+      var actualKey: Int = 0
+      var pos: Int = 0
+
+      def hasNext(): Boolean = keysCopy.hasNext
+
+      def next(): E = {
+        val k = keysCopy.next()
+        val v = inner(k)
+        actualKey = k
+        pos+=1
+        v
+      }
+
+      def remove(): Unit = 
+        if (pos > 0) inner.remove(actualKey)
+    }
+
+  private def removeWhere[T](c: Collection[T])(p: T => Boolean): Boolean = {
```

Comment:
Why is the structure of this `removeWhere` different than the others? Am I missing something?